lib/internal/test_runner/mock/fake_timers.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const {
+  DateNow,
+  FunctionPrototypeBind,
+  globalThis,
+  Promise,
+} = primordials;
+
+const PriorityQueue = require('internal/priority_queue');
+const nodeTimers = require('timers');
+const nodeTimersPromises = require('timers/promises');
+
+function compareTimersLists(a, b) {
+  return (a.runAt - b.runAt) || (a.id - b.id);
+}
+
+function setPosition(node, pos) {
+  node.priorityQueuePosition = pos;
+}
+
+
+class FakeTimers {
+  #realSetTimeout;
+  #realClearTimeout;
+  #realSetInterval;
+  #realClearInterval;
+
+  #isEnabled = false;
+  #currentTimer = 1;
+  #fakeTimers = {};
+  #now = DateNow();
+
+  #timers = new PriorityQueue(compareTimersLists, setPosition);
+
+  #setTimeout = FunctionPrototypeBind(this.#createTimer, this, false);
+  #clearTimeout = FunctionPrototypeBind(this.#clearTimer, this);
+  #setInterval = FunctionPrototypeBind(this.#createTimer, this, true);
+  #clearInterval = FunctionPrototypeBind(this.#clearTimer, this);
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = this.#currentTimer++;
+    this.#timers.insert({
+      id: timerId,
+      callback,
+      runAt: DateNow() + delay,
+      interval: isInterval,
+      args,
+    });
+
+    return timerId;
+  }
+
+  #clearTimer(position) {
+    this.#timers.removeAt(position);
+  }
+
+  tick(time = 1) {
+
+    // if (!this.isEnabled) {
+    //   throw new Error('you should enable fakeTimers first by calling the .enable function');
+    // }
+    this.#now += time;
+    const timers = this.#timers;
+
+    for (let timer = timers.peek(); timer?.runAt <= this.#now; timer = timers.peek()) {
```

Comment:
can `timer` be null? if so, `timer?.runAt` might be true unexpectedly

Reply:
@ljharb how?

Reply:
`null <= 123` is true, for example, and `null?.runAt <= Date.now()` would thus always be true.

If `timer` can't ever be null, then it's not an issue.

Reply:
Ah, good catch!

Reply:
thanks @ljharb